tensorflow/compiler/jit/tests/auto_clustering_test.cc

```diff
@@ -29,13 +32,41 @@ class AutoClusteringTestImpl : public AutoClusteringTest {
         absl::StrCat(file_name_without_extension, ".pbtxt"),
         absl::StrCat(file_name_without_extension, ".golden_summary"));
   }
+
+  // Decompress file ${key}.pbtxt.gz into ${key}.pbtxt
+  // and test auto clustering with the .pbtxt.
+  Status RunAutoClusteringTestWithGzippedPbtxt(absl::string_view key);
 };
 
-Status BenchmarkHelper(absl::string_view key, benchmark::State& state) {
-  return BenchmarkMarkForCompilation(
-      absl::StrCat(testing::TensorFlowSrcRoot(), "/compiler/jit/tests/", key,
-                   ".pbtxt"),
-      state);
+Status AutoClusteringTestImpl::RunAutoClusteringTestWithGzippedPbtxt(
+    absl::string_view key) {
+  string file_name_without_extension =
+      absl::StrCat(testing::TensorFlowSrcRoot(), "/compiler/jit/tests/", key);
+  string input_fname = absl::StrCat(file_name_without_extension, ".pbtxt.gz");
+  string pbtxt_fname = absl::StrCat(file_name_without_extension, ".pbtxt");
+  string summary_fname =
+      absl::StrCat(file_name_without_extension, ".golden_summary");
+
+  Env* env = Env::Default();
+  std::unique_ptr<RandomAccessFile> file_reader;
+  TF_RETURN_IF_ERROR(env->NewRandomAccessFile(input_fname, &file_reader));
+  std::unique_ptr<io::RandomAccessInputStream> input_stream(
+      new io::RandomAccessInputStream(file_reader.get()));
+  constexpr int k_buffer_size = 256 << 10;  // 256kb
+  io::ZlibInputStream in(input_stream.get(),
+                         /*input_buffer_bytes=*/k_buffer_size,
+                         /*output_buffer_bytes=*/k_buffer_size,
+                         io::ZlibCompressionOptions::GZIP());
+  string decompressed_data;
+  Status s = in.ReadNBytes(INT_MAX, &decompressed_data);
+  if (!s.ok() && !errors::IsOutOfRange(s)) {
+    // OutOfRange is fine since we set the number of read bytes to INT_MAX.
+    // Only return other kinds of errors.
+    return s;
+  }
+  TF_RETURN_IF_ERROR(WriteStringToFile(env, pbtxt_fname, decompressed_data));
```

Comment:
You've probably already thought of this, but how difficult would it be to avoid creating a temporary file here?  Can we directly parse a protobuf out of `decompressed_data`?

Reply:
Require some work and need to change some interface of `RunAutoClusteringTest()`, but it should not be too difficult. Will make the change to avoid the temp file.